packages/react-dom/src/__tests__/ReactUpdates-test.js

```diff
@@ -48,306 +48,395 @@ describe('ReactUpdates', () => {
     );
   }
 
-  it('should batch state when updating state twice', () => {
-    let updateCount = 0;
-
-    class Component extends React.Component {
-      state = {x: 0};
-
-      componentDidUpdate() {
-        updateCount++;
-      }
+  it('should batch state when updating state twice', async () => {
+    let componentState;
+    let setState;
+
+    function Component() {
+      const [state, _setState] = React.useState(0);
+      componentState = state;
+      setState = _setState;
+      React.useLayoutEffect(() => {
+        Scheduler.log('Commit');
+      });
 
-      render() {
-        return <div>{this.state.x}</div>;
-      }
+      return <div>{state}</div>;
     }
 
-    const instance = ReactTestUtils.renderIntoDocument(<Component />);
-    expect(instance.state.x).toBe(0);
+    const container = document.createElement('div');
+    const root = ReactDOMClient.createRoot(container);
+    await act(() => {
+      root.render(<Component />);
+    });
 
-    ReactDOM.unstable_batchedUpdates(function () {
-      instance.setState({x: 1});
-      instance.setState({x: 2});
-      expect(instance.state.x).toBe(0);
-      expect(updateCount).toBe(0);
+    assertLog(['Commit']);
+    expect(container.firstChild.textContent).toBe('0');
+
+    await act(() => {
+      setState(1);
+      setState(2);
+      expect(componentState).toBe(0);
+      expect(container.firstChild.textContent).toBe('0');
+      assertLog([]);
     });
 
-    expect(instance.state.x).toBe(2);
-    expect(updateCount).toBe(1);
+    expect(componentState).toBe(2);
+    assertLog(['Commit']);
+    expect(container.firstChild.textContent).toBe('2');
   });
 
-  it('should batch state when updating two different state keys', () => {
-    let updateCount = 0;
+  it('should batch state when updating two different states', async () => {
+    let componentStateA;
+    let componentStateB;
+    let setStateA;
+    let setStateB;
 
-    class Component extends React.Component {
-      state = {x: 0, y: 0};
+    function Component() {
+      const [stateA, _setStateA] = React.useState(0);
+      const [stateB, _setStateB] = React.useState(0);
+      componentStateA = stateA;
+      componentStateB = stateB;
+      setStateA = _setStateA;
+      setStateB = _setStateB;
 
-      componentDidUpdate() {
-        updateCount++;
-      }
+      React.useLayoutEffect(() => {
+        Scheduler.log('Commit');
```

Comment:
I also ported any tests tracking logs/ops to use the newer `Scheduler.log` test helpers to modernize them.